recipes/open62541/all/conanfile.py

```diff
@@ -0,0 +1,289 @@
+            if self.options.subscription_events:
+                raise ConanInvalidConfiguration(
+                    "Open62541 requires subscription option")
+
+        if not self.options.discovery:
+            if self.options.discovery_multicast or self.options.discovery_semaphore:
+                raise ConanInvalidConfiguration(
+                    "Open62541 requires discovery option")
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+        if not self.options.cpp_compatible:
+            del self.settings.compiler.cppstd
+            del self.settings.compiler.libcxx
+
+    def source(self):
+        archive_name = self.name + "-" + self.version
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename(archive_name, self._source_subfolder)
+
+        submodule_filename = os.path.join(
+            os.path.dirname(__file__), 'submoduledata.yml')
+        with open(submodule_filename, 'r') as submodule_stream:
+            submodules_data = yaml.load(submodule_stream)
+            for path, submodule in submodules_data["submodules"][self.version].items():
+                filename = os.path.basename(submodule["url"])
+                archive_name = submodule["archive_pattern"].format(
+                    version=os.path.splitext(filename.replace('v', ''))[0])
+
+                submodule_data = {
+                    "url": submodule["url"],
+                    "sha256": submodule["sha256"]
+                }
+
+                tools.get(**submodule_data)
+                submodule_source = os.path.join(self._source_subfolder, path)
+                tools.rmdir(submodule_source)
+                os.rename(archive_name, submodule_source)
+
+    def _get_log_level(self):
+        return {
+            "Fatal": "600",
+            "Error": "500",
+            "Warning": "400",
+            "Info": "300",
+            "Debug": "200",
+            "Trace": "100",
+            "PackageOption": "300"
+        }.get(str(self.options.logging_level), "300")
+
+    def _get_multithreading_option(self):
+        return {
+            "Off": "0",
+            "Threadsafe": "100",
+            "Internal threads": "200"
+        }.get(str(self.options.multithreading), "0")
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+
+        if self.settings.compiler == "clang" and not self.options.shared:
+            raise ConanInvalidConfiguration(
+                "Clang compiler can not be used to build a static library")
+
+        self._cmake = CMake(self)
+        self._cmake.verbose = True
+
+        version = Version(self.version)
+        self._cmake.definitions["OPEN62541_VER_MAJOR"] = version.major(
+            fill=False)
+        self._cmake.definitions["OPEN62541_VER_MINOR"] = version.minor(
+            fill=False)
+        self._cmake.definitions["OPEN62541_VER_PATCH"] = version.patch()
```

Comment:
This is usually handled by CMake, can you confirm it's required?

Reply:
https://github.com/conan-io/conan-center-index/pull/3786/files#diff-a901f6e581d3a2c7fec1494a71e01af82183b2cc90194c60fe566e1cef6c981bR23

Cool!

Reply:
sadly Open62541 uses non standard version definitions in their `CMakeLists.txt` so I have to explicitly set then in my configure method 
>> for their binary releases, they use git to generate `OPEN62541_VERSION` variable which is made up from these parts, furthermore these definitions are used in source code. So to avoid headaches down the line I set the version as it would be set by the CMake git module tool from Open62541)